provider/azure/environprovider.go

```diff
@@ -97,6 +98,12 @@ func (prov *azureEnvironProvider) Open(args environs.OpenParams) (environs.Envir
 	return environ, nil
 }
 
+// CloudSchema returns the schema used to validate input for add-cloud.  Since
+// this provider does not support custom clouds, this always returns nil.
+func (p azureEnvironProvider) CloudSchema() *jsonschema.Schema {
```

Comment:
It doesn't look to me like any of these provider changes (CloudSchema implementations) are tested?